primitives/runtime/src/generic/header.rs

```diff
@@ -81,13 +83,16 @@ where
 		+ MaybeSerializeDeserialize
 		+ MaybeFromStr
 		+ Debug
+		+ Default
 		+ sp_std::hash::Hash
 		+ MaybeDisplay
 		+ AtLeast32BitUnsigned
-		+ Codec
+		+ FullCodec
 		+ Copy
+		+ MaxEncodedLen
 		+ Into<U256>
-		+ TryFrom<U256>,
+		+ TryFrom<U256>
+		+ TypeInfo,
```

Comment:
double checking that these are all needed?

Reply:
All in all, it is worth acknowledging that what we are forced to do here is pretty sub-optimal in many ways. We are moving bounds that frame-system imposed on these types to here. 

That aside though, the outcome seems fine. We already make pretty strong assumptions on all types being `Codec` and `TypeInfo`. I find `MaxEncodedLen` a bit more iffy, as a runtime/FRAME-alternative that is not for parachains would probably not care about this at all. 

At the same time, it is easy enough to provide mock implementation to all of this, so not an issue.

Reply:
Yes, verified once again that these are needed.